js/src/behaviors/segmented-controls.ts

```diff
@@ -0,0 +1,140 @@
+	}
+}
+
+function initSegmentedControlNavClickListeners() {
+	window.addEventListener('click', (event: Event) => {
+		const target =
+			event.target instanceof Element &&
+			(event.target.closest('[data-segmented-control-nav]') as HTMLButtonElement);
+
+		if (!target) {
+			return;
+		}
+
+		event.preventDefault();
+		const container = event.target.closest('[data-segmented-controls-container]') as HTMLElement;
+		const segments = Array.from(
+			container.querySelectorAll('[data-segmented-control]')
+		) as HTMLButtonElement[];
+
+		// get current activated tab
+		const currentActiveButton = container.querySelector(
+			'[aria-selected="true"]'
+		) as HTMLButtonElement;
+		const index = parseInt(currentActiveButton?.dataset.segmentedControl as string);
+
+		// update tab
+		if (index > 1 && target.dataset.segmentedControlNav === 'previous') {
+			updateSegmentedControlState(container, currentActiveButton, index - 1);
+			updateSegmentedControlItem(container, index, index - 1);
+		} else if (index < segments.length && target.dataset.segmentedControlNav === 'next') {
+			updateSegmentedControlState(container, currentActiveButton, index + 1);
+			updateSegmentedControlItem(container, index, index + 1);
+		}
+		updateSegmentedControlNav(container);
+	});
+}
+
+function initSegmentedControlClickListeners() {
+	window.addEventListener('click', (event: Event) => {
+		const target =
+			event.target instanceof Element &&
+			(event.target.closest('[data-segmented-control]') as HTMLButtonElement);
+
+		if (!target) {
+			return;
+		}
+
+		event.preventDefault();
+		const container = event.target.closest('[data-segmented-controls-container]') as HTMLElement;
+		updateSegmentedControlNav(container);
+	});
+	window.addEventListener('keydown', (event: KeyboardEvent) => {
+		if (event.key !== 'ArrowLeft' && event.key !== 'ArrowRight') {
+			return;
+		}
+
+		const target =
+			event.target instanceof Element &&
+			(event.target.closest('[data-segmented-control]') as HTMLButtonElement);
+
+		if (!target) {
+			return;
+		}
+
+		event.preventDefault();
+		const container = event.target.closest('[data-segmented-controls-container]') as HTMLElement;
+		updateSegmentedControlNav(container);
+	});
+}
+
+function updateSegmentedControlNav(container: HTMLElement = document.body) {
+	const segments = Array.from(
+		container.querySelectorAll('[data-segmented-control]')
+	) as HTMLButtonElement[];
+	const previousButton = container.querySelector(
+		'[data-segmented-control-nav="previous"]'
+	) as HTMLButtonElement;
+	const nextButton = container.querySelector(
+		'[data-segmented-control-nav="next"]'
+	) as HTMLButtonElement;
+	const currentActiveButton = container.querySelector(
+		'[aria-selected="true"]'
+	) as HTMLButtonElement;
+	const index = parseInt(currentActiveButton?.dataset.segmentedControl as string);
+
+	// slide active button into view
+	currentActiveButton.scrollIntoView({ behavior: 'auto', block: 'nearest', inline: 'center' });
```

Comment:
I'm not sure if this is the right place for this feedback, but I'm seeing a weird issue where at some browser sizes, selecting later tabs causes the page layout to jump left.

```
Browser: Microsoft Edge
Browser height: 976px
Browser width: 1826px
```


https://user-images.githubusercontent.com/18060369/214386348-8c384a3b-65f1-4cca-b602-5cb15d537ac9.mp4

I suspect this has to do with `scrollIntoView` here. Notably, when the layout jumps, I'm not able to scroll horizontally to a point where the resembles how it was to start - I have to select an earlier tab to get back.

Reply:
weird, i am unable to repo this

Reply:
@wibjorn @ollips @homing1 Are any of y'all able to reproduce this? If not, we can resolve this.

Reply:
I tried different screen sizes, but also unable to reproduce. The only thing that jumps for me is the segmented control itself when the previous/next button appear/disappear.

Reply:
I was able to repo! But I had to change the zoom level to 90%

Reply:
Can no longer reproduce this, so I'm resolving this - let's definitely keep an eye on it, though, just in case.